app/frontend/wallet/account-manager.ts

```diff
@@ -0,0 +1,65 @@
+import NamedError from '../helpers/NamedError'
+import {Account} from './account'
+import {MAX_ACCOUNT_COUNT} from './constants'
+
+const AccountManager = ({config, cryptoProvider, blockchainExplorer}) => {
+  const accounts: Array<ReturnType<typeof Account>> = []
+
+  function getAccount(accountIndex: number) {
+    return accounts[accountIndex]
+  }
+
+  function discoverNewAccount() {
+    return Account({
+      config,
+      cryptoProvider,
+      blockchainExplorer,
+      accountIndex: accounts.length,
+    })
+  }
+
+  async function addNewAccount(account) {
+    await account.isAccountUsed() // To ensure user exported pubkey
```

Comment:
why not ensuring that the key is exported right in `discoverNewAccount`? And if that's really a concern, then wither the `Account()` function itself should take care of that or if that would cause issues, then there should be an `init()` function that would do that and not just "abuse" a seemingly random function for that purpose

Reply:
You are right, I tried to not "abuse" the function just for this purpose but we need to ensure that public key is exported before adding the account to the list. I didn't want to create an init function just for this purpose and thought that the comment would suffice but you are right, the init function is cleaner. Actually, the function is needed only in `exploreNewAccount` function because `discoverAccounts` function checks it with `isAccountUsed()`. 

I just wanted to ensure that `addNewAccount(account)` and pubKey export happen in one place.